nilearn/plotting/surface/_matplotlib_backend.py

```diff
@@ -494,3 +508,102 @@ def _plot_surf(
+
+        if axes.name != "3d":
+            raise ValueError("Axes must be 3D.")
+
+        # test if axes contains Poly3DCollection, if not initialize surface
+        if not axes.collections or not isinstance(
+            axes.collections[0], Poly3DCollection
+        ):
+            _ = self._plot_surf(surf_mesh, hemi=hemi, axes=axes, **kwargs)
+
+        if levels is None:
+            levels = np.unique(roi_map)
+
+        if labels is None:
+            labels = [None] * len(levels)
+
+        if colors is None:
+            n_levels = len(levels)
+            vmax = n_levels
+            cmap = plt.get_cmap(cmap)
+            norm = Normalize(vmin=0, vmax=vmax)
+            colors = [cmap(norm(color_i)) for color_i in range(vmax)]
+        else:
+            try:
+                colors = [to_rgba(color, alpha=1.0) for color in colors]
+            except ValueError:
+                raise ValueError(
+                    "All elements of colors need to be either a"
+                    " matplotlib color string or RGBA values."
+                )
+
+        if not (len(levels) == len(labels) == len(colors)):
+            raise ValueError(
+                "Levels, labels, and colors "
+                "argument need to be either the same length or None."
+            )
+
+        _, faces = self.load_surf_mesh(surf_mesh)
+        roi = self.load_surf_data(roi_map)
+
+        patch_list = []
+        for level, color, label in zip(levels, colors, labels):
+            roi_indices = np.where(roi == level)[0]
+            faces_outside = get_faces_on_edge(faces, roi_indices)
+            # Fix: Matplotlib version 3.3.2 to 3.3.3
+            # Attribute _facecolors3d changed to _facecolor3d in
+            # matplotlib version 3.3.3
+            if compare_version(mpl_version, "<", "3.3.3"):
+                axes.collections[0]._facecolors3d[faces_outside] = color
+                if axes.collections[0]._edgecolors3d.size == 0:
+                    axes.collections[0].set_edgecolor(
+                        axes.collections[0]._facecolors3d
+                    )
+                axes.collections[0]._edgecolors3d[faces_outside] = color
+            else:
+                axes.collections[0]._facecolor3d[faces_outside] = color
+                if axes.collections[0]._edgecolor3d.size == 0:
+                    axes.collections[0].set_edgecolor(
+                        axes.collections[0]._facecolor3d
+                    )
+                axes.collections[0]._edgecolor3d[faces_outside] = color
+            if label and legend:
+                patch_list.append(Patch(color=color, label=label))
+        # plot legend only if indicated and labels provided
+        if legend and np.any([lbl is not None for lbl in labels]):
+            figure.legend(handles=patch_list)
+            # if legends, then move title to the left
+        if title is None and hasattr(figure._suptitle, "_text"):
+            title = figure._suptitle._text
```

Comment:
that's a legit coverage complaint: see the updloaded HTML coverage report

https://github.com/nilearn/nilearn/actions/runs/14991516206/artifacts/3112733623